src/Analyzers/CSharp/Analyzers/SimplifyPropertyPattern/SimplifyPropertyPatternHelpers.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.CodeAnalysis.CSharp.SimplifyPropertyPattern
+{
+    internal static class SimplifyPropertyPatternHelpers
+    {
+        public static bool IsSimplifiable(
+            SubpatternSyntax subpattern,
+            [NotNullWhen(true)] out SubpatternSyntax? innerSubpattern)
+        {
+            // can't simplify if we have anything inside other than a property pattern clause.  i.e.
+            // `a: { b: ... } x` is not simplifiable as we'll lose the `x` binding for the `a` property.
+            //
+            // can't simplify `a: { }` or `a: { b: ..., c: ... }`
+            if (subpattern is
+                {
+                    ExpressionColon: { } outer,
+                    Pattern: RecursivePatternSyntax
+                    {
+                        Type: null,
+                        PositionalPatternClause: null,
+                        Designation: null,
```

Comment:
I saw tests with designation part, but not with type or positional part. Consider covering that too

Reply:
will do!